packages/owl-bot/src/fetch-configs.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AdmZip from 'adm-zip';
+import {OctokitType} from './octokit-util';
+import tmp from 'tmp';
+import {OwlBotLock} from './config-files';
+import {collectConfigs, OwlBotYamlAndPath} from './configs-store';
+import * as fs from 'fs';
+import path from 'path';
+
+/**
+ * Fetches the configuration files from a github repo.
+ */
+export async function fetchConfigs(
+  octokit: OctokitType,
+  githubRepo: {
+    owner: string;
+    repo: string;
+    ref: string;
+  }
+): Promise<[OwlBotLock | undefined, OwlBotYamlAndPath[]]> {
```

Comment:
I like that the zip logic is encapsulated in a helper file, makes it easier to swap out if we ever wanted to.